CASMHMS-5602 HMS Helm CT test documentation updates (#1988)
Cray-HPE/docs-csm: operations/validate_csm_health.md

On CT test errors or failures, the script will print the path to the CT test log file for the administrator to inspect.
If one or more failures occur, investigate the cause of each and take remediation steps if needed.
See the [Interpreting HMS Health Check Results](../troubleshooting/interpreting_hms_health_check_results.md) documentation for more information.

After remediating a test failure for a particular service, just the tests for that individual service
can be re-run by supplying the name of the service to the `run_hms_ct_tests.sh` script with the `-t` option:

```bash
/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh -t <service>
```

To list the HMS services that can be tested, use the `-l` option:

```bash
/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh -l
```

### 2.2 Hardware State Manager discovery validation

By the time the CSM health validation is first performed on a system, the Hardware State Manager (HSM)
should have completed its discovery of the system. This section provides steps to verify
that discovery has completed successfully and consists of two steps.

> 

Reviewer: ```suggestion
```

> Run the HMS CT tests. This is done by running the `run_hms_ct_tests.sh` script:

Reviewer: ```suggestion
(`ncn-mw#`) Run the HMS CT tests.
```

> ```text

Reviewer 2: Isn't this ran in bash?

Author: Some of these are commands that run in the terminal but aren't technically valid BASH, like the <service> part of:
`/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh -t <service>`

From what I can tell the only difference in the code blocks between `text` and `bash` are that colorations are attempted to be set when using `bash` and they get set incorrectly or in a potentially confusing manner in some cases. I opted to use `text` since the colorations seem unnecessary and might do more harm than good in these examples.

Author: That being said, it's just my preference so if we are aiming to make this standardized across the documentation I can change them to `bash`.

> ```text

Reviewer: ```suggestion
```bash
```

/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh
```

The return code of the script is zero if all HMS CT tests run and pass, non-zero if not.

> On CT test errors or failures the script will print the path to the CT test log file for the admin to inspect.

Reviewer: ```suggestion
On CT test errors or failures, the script will print the path to the CT test log file for the administrator to inspect.
```

If one or more failures occur, investigate the cause of each and take remediation steps if needed.
See the [Interpreting HMS Health Check Results](../troubleshooting/interpreting_hms_health_check_results.md) documentation for more information.

After remediating a test failure for a particular service, just the tests for that individual service
can be re-run by supplying the name of the service to the `run_hms_ct_tests.sh` script with the `-t` option:

> ```text

Reviewer: ```suggestion
```bash
```

/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh -t <service>
```

To list the HMS services that can be tested, use the `-l` option:

> ```text

Reviewer: ```suggestion
```bash
```

/opt/cray/csm/scripts/hms_verification/run_hms_ct_tests.sh -l
```

### 2.2 Hardware State Manager discovery validation

By this point in the installation process, the Hardware State Manager (HSM) should
have completed its discovery of the system. This section provides steps to verify

> that discovery has completed successfully and consists of two parts. First, that all

Reviewer: ```suggestion
that discovery has completed successfully and consists of two parts:

```

> hardware attempted to be discovered by HSM was successful, and second, that all of

Reviewer: ```suggestion
1. All hardware attempted to be discovered by HSM was successfully discovered.
```

> the expected hardware in the system is present in HSM.

Reviewer: ```suggestion
1. All hardware that is expected to be in the system is present in HSM.
```

> To verify that discovery completed successfully and that Redfish endpoints for the

Reviewer: ```suggestion
(`ncn-mw#`) To verify that discovery completed successfully and that Redfish endpoints for the
```

system hardware have been populated in HSM, run the `hsm_discovery_status_test.sh`

> script on a worker or master NCN:

Reviewer: ```suggestion
script.
```

> ```text

Reviewer: ```suggestion
```bash
```

/opt/cray/csm/scripts/hms_verification/hsm_discovery_status_test.sh
```

The script will return an exit code of zero if there are no failures. Otherwise, the
script will return a non-zero exit code along with output indicating which components
failed discovery and troubleshooting steps for determining why discovery failed.

> Next, to verify that all of the expected hardware in the system is present in HSM, a

Reviewer: ```suggestion
1. (`ncn-mw#`) To verify that all of the expected hardware in the system is present in HSM, a
```

comparison needs to be made between HSM and the System Layout Service (SLS) which
provides the foundational information for the hardware that makes up the system.

To perform this comparison, execute the `verify_hsm_discovery.py` script on a worker
or master NCN. The result is pass/fail (returns zero or non-zero):

```text
/opt/cray/csm/scripts/hms_verification/verify_hsm_discovery.py
```

> The output will ideally appear as follows, if there are mismatches these will be

Reviewer: ```suggestion
The output will ideally appear as follows; if there are mismatches, then these will be
```

displayed in the appropriate section of the output. Refer to
[2.2.1 Interpreting results](#221-interpreting-hsm-discovery-results) and

> [2.2.2 Known Issues](#222-known-issues-with-hsm-discovery-validation) below to

Reviewer: ```suggestion
[2.2.2 Known Issues](#222-known-issues-with-hsm-discovery-validation) in order to
```
